src/reyna/Blackout/BlackoutTime.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Reyna
+{
+    using System;
+    using System.Globalization;
+
+    public class BlackoutTime
+    {
+        public bool CanSendAtTime(DateTime now, string ranges)
+        {
+            if (String.IsNullOrEmpty(ranges))
+            {
+                return true;
+            }
+
+            string[] rangesSplit = ranges.Split(',');
+            foreach (string range in rangesSplit)
+            {
+                TimeRange timeRange = this.ParseTime(range);
+                Time timeNow = new Time(now.Hour, now.Minute);
+                if (timeRange.Contains(timeNow) && !timeRange.IsEmpty())
```

Comment:
In Android we moved the logic of isEmpty inside contains, because it does not make sense that a timeRange contains a time-point and is empty at the same time. Should do the same here.